crates/core/agent_mapper_interface/src/topic.rs

```diff
@@ -22,47 +27,47 @@ impl IncomingTopic {
 }
 
 impl TryFrom<String> for IncomingTopic {
-    type Error = MapperTopicError;
+    type Error = TopicError;
 
     fn try_from(value: String) -> Result<Self, Self::Error> {
         match value.as_str() {
             r#"tedge/commands/res/software/list"# => Ok(IncomingTopic::SoftwareListResponse),
             r#"tedge/commands/res/software/update"# => Ok(IncomingTopic::SoftwareUpdateResponse),
             r#"c8y/s/ds"# => Ok(IncomingTopic::SmartRestRequest),
             r#"tedge/commands/res/control/restart"# => Ok(IncomingTopic::RestartResponse),
-            err => Err(MapperTopicError::UnknownTopic {
+            err => Err(TopicError::UnknownTopic {
                 topic: err.to_string(),
             }),
         }
     }
 }
 
 impl TryFrom<&str> for IncomingTopic {
-    type Error = MapperTopicError;
+    type Error = TopicError;
 
     fn try_from(value: &str) -> Result<Self, Self::Error> {
         Self::try_from(value.to_string())
     }
 }
 
 impl TryFrom<Topic> for IncomingTopic {
-    type Error = MapperTopicError;
+    type Error = TopicError;
 
     fn try_from(value: Topic) -> Result<Self, Self::Error> {
         value.name.try_into()
     }
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub(crate) enum OutgoingTopic {
+pub enum OutgoingTopic {
```

Comment:
The name is misleading after the move from the mapper to this agent interface.

```suggestion
pub enum RequestTopic {
```

Reply:
done